### ISSUE TITLE
Way to programmatically parse .razor (blazor) files and get an object model

### ISSUE THREAD

**Issue by b-giavotto** [dotnet/aspnetcore#48449]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

What should be useful will be the ability to parse a .razor/blazor file and get an object model describing it, like could be achieved with xml or html files. However due to some differences in markup syntax ( the tag attributes allow to embed c# expressions which have themselves double quoted strings, for example) it is not possible to use such kind of parser.

Does a similar api or library already exists? how could be achieved?
Basically what i need to do is 
a)Load a blazor .razor file
b)parse and get an object model of the .razor document
c)manipulate elements via (bl)(r)azor DOM object model
d) write back a text .razor file.

### Describe the solution you'd like

see above

**Comment by javiercn** [comment]
@b-giavotto thanks for contacting us.

We don't have plans to offer this functionality, this amounts to exposing all the compiler internals in a similar way to what Roslyn does and is not something that we plan to prioritize in the near future. The compiler bits are on dotnet/razor and you can play with them at your own risk, but beware that nothing in there is public API.
